neutron_vpnaas/services/vpn/service_drivers/ovn_ipsec.py

```diff
@@ -407,7 +408,49 @@ def update_ipsec_site_connection(
             context, old_ipsec_site_connection, ipsec_site_connection)
 
 
-class IPsecOvnVpnAgentApi(base_ipsec.IPsecVpnAgentApi):
+class IPsecOvnVpnAgentApi(object):
```

Comment:
The only thing you want to do is rewrite the_agent_notification function, right? you can still inherit from base_ipsec.IPsecVpnAgentApi, then override _agent_notification instead of copy all the code here.